web/packages/teleport/src/Users/Users.tsx

```diff
@@ -86,21 +94,87 @@ export function Users(props: State) {
     params: { search },
   });
 
+  // Cleanup controller on unmount
   useEffect(() => {
-    // Cancel previous request and create new controller
     abortControllerRef.current?.abort();
     abortControllerRef.current = new AbortController();
 
     serverSidePagination.fetch();
   }, [search]);
 
-  // Cleanup controller on unmount
   useEffect(() => {
     return () => {
       abortControllerRef.current?.abort();
     };
   }, []);
 
+  const selectedUser =
+    username && serverSidePagination.attempt.status === 'success'
+      ? serverSidePagination.fetchedData.agents.find(
+          u => u.name === username
+        ) || null
+      : null;
```

Comment:
This expression is overly complex and I have a hard time comprehending it. Can we somehow split it or extract to a function and use more imperative constructs?